tests/components/powerwall/test_binary_sensor.py

```diff
@@ -0,0 +1,60 @@
+"""The binary sensor tests for the powerwall platform."""
+
+from asynctest import patch
+
+from homeassistant.components.powerwall.const import DOMAIN
+from homeassistant.const import STATE_ON
+from homeassistant.setup import async_setup_component
+
+from .mocks import _mock_get_config, _mock_powerwall_with_fixtures
+
+
+async def test_sensors(hass):
+    """Test creation of the binary sensors."""
+
+    mock_powerwall = await _mock_powerwall_with_fixtures(hass)
+
+    with patch(
+        "homeassistant.components.powerwall.config_flow.PowerWall",
+        return_value=mock_powerwall,
+    ), patch(
+        "homeassistant.components.powerwall.PowerWall", return_value=mock_powerwall,
+    ):
+        assert await async_setup_component(hass, DOMAIN, _mock_get_config())
+        await hass.async_block_till_done()
+
+    entity = hass.states.get("binary_sensor.grid_status")
```

Comment:
This is a state that we get here, so we should call it so.
```suggestion
    state = hass.states.get("binary_sensor.grid_status")
```